Library/Homebrew/dependencies_helpers.rb

```diff
@@ -17,19 +17,28 @@ def args_includes_ignores(args)
     ignores << :recommended? if args.skip_recommended?
     ignores << :satisfied? if args.missing?
 
-    [includes, ignores]
+    recursive_ignores = []
+    recursive_ignores << :build? if args.respond_to?(:direct_build?) && args.direct_build?
+    recursive_ignores << :test? if args.respond_to?(:direct_test?) && args.direct_test?
+
+    [includes, ignores, recursive_ignores]
   end
 
-  def recursive_includes(klass, root_dependent, includes, ignores)
+  def recursive_includes(klass, root_dependent, includes, ignores, recursive_ignores: nil, skip: nil)
     raise ArgumentError, "Invalid class argument: #{klass}" if klass != Dependency && klass != Requirement
 
     cache_key = "recursive_includes_#{includes}_#{ignores}"
+    cache_key = "#{cache_key}_#{recursive_ignores}" if recursive_ignores.present?
+    cache_key = "#{cache_key}_#{skip}" if skip.present?
```

Comment:
I wonder if it's really still useful to have a cache key that's this complex to create.